client/finality-grandpa/src/finality_proof.rs

```diff
@@ -146,11 +147,14 @@ pub enum FinalityProofError {
 	Client(#[from] sp_blockchain::Error),
 }
 
-fn prove_finality<Block, B>(
+/// Prove finality for the given block number by returning a Justification for the last block of
+/// the authority set.
+pub fn prove_finality<Block, B>(
```

Comment:
This should be kept private and you should add a function to `SharedAuthoritySet`.

Please also document what `collect_unknown_headers` does.

Reply:
Ok, maybe user should impl this logic outside substrate by theirself.

Reply:
Maybe I should close this PR.